src/libs/actions/Report.js

```diff
@@ -421,6 +426,40 @@ function navigateToAndOpenReport(userLogins) {
     Navigation.navigate(ROUTES.getReportRoute(reportID));
 }
 
+/**
+ * This will navigate to an existing thread, or create a new one if necessary
+ *
+ * @param {String} childReportID The reportID we are trying to open
+ * @param {Object} parentReportAction the parent comment of a thread
+ * @param {String} parentReportID The reportID of the parent
+ *
+ */
+function navigateToAndOpenChildReport(childReportID = '0', parentReportAction = {}, parentReportID = '0') {
+    if (childReportID !== '0') {
+        openReport(childReportID);
+        Navigation.navigate(ROUTES.getReportRoute(childReportID));
+    } else {
+        const participants = _.uniq([currentUserEmail, parentReportAction.actorEmail]);
+        const formattedUserLogins = _.map(participants, login => OptionsListUtils.addSMSDomainIfPhoneNumber(login).toLowerCase());
+        const newChat = ReportUtils.buildOptimisticChatReport(
+            formattedUserLogins,
+            lodashGet(parentReportAction, ['message', 0, 'text']),
+            '',
+            CONST.POLICY.OWNER_EMAIL_FAKE,
+            CONST.POLICY.OWNER_EMAIL_FAKE,
+            false,
+            '',
+            undefined,
+            CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS,
+            parentReportAction.reportActionID,
+            parentReportID,
```

Comment:
I need to test that these are correct

Reply:
@chiragsalian should the child report inherit the chat type / workspace room info from the parent?